python/cudf_polars/cudf_polars/containers/dataframe.py

```diff
@@ -108,7 +114,10 @@ def to_polars(self) -> pl.DataFrame:
             _create_polars_column_metadata(name, dtype.polars_type)
             for name, dtype in zip(name_map, self.dtypes, strict=True)
         ]
-        table_with_metadata = _ObjectWithArrowMetadata(self.table, metadata)
+        table_with_metadata = _ObjectWithArrowMetadata(
+            self.table, metadata, self.stream
+        )
+        self.stream.synchronize()
```

Comment:
I think this sync should now not be necessary, `_to_host_array` copies on `stream` and then syncs the stream.

Reply:
Thanks, I was wondering about that.